lnrpc/signrpc/signer_server.go

```diff
@@ -875,6 +875,41 @@ func (s *Server) MuSig2CombineKeys(_ context.Context,
 	}, nil
 }
 
+// secNonceToPubNonce takes our two secret nonces, and produces their two
+// corresponding EC points, serialized in compressed format.
+//
+// NOTE: This was copied from btcsuite/btcec/musig2/nonces.go
+func secNonceToPubNonce(secNonce [musig2.SecNonceSize]byte,
+) [musig2.PubNonceSize]byte {
+
+	var k1Mod, k2Mod btcec.ModNScalar
+	k1Mod.SetByteSlice(secNonce[:btcec.PrivKeyBytesLen])
+	k2Mod.SetByteSlice(secNonce[btcec.PrivKeyBytesLen:])
+
+	var r1, r2 btcec.JacobianPoint
+	btcec.ScalarBaseMultNonConst(&k1Mod, &r1)
+	btcec.ScalarBaseMultNonConst(&k2Mod, &r2)
+
+	// Next, we'll convert the key in jacobian format to a normal public
+	// key expressed in affine coordinates.
+	r1.ToAffine()
+	r2.ToAffine()
+	r1Pub := btcec.NewPublicKey(&r1.X, &r1.Y)
+	r2Pub := btcec.NewPublicKey(&r2.X, &r2.Y)
```

Comment:
Nit: I'm somewhat surprised that the singular version of this isn't its own function.

Reply:
Just mapping a secret nonce to a public nonce? It's in the `musig2` package, but unadvertised. Reason is that it tries to force a "safe" API by always getting in strong randomness, then applying all the musig2-bip specific item like: appending the public key to the secret nonces itself.

Reply:
Could've exported it, but then would've had to commit to that addition to the API.

Reply:
Yeah I guess what I see here is a function that takes a full secnonce encoding, breaks it into two secret u256's and converts them to ECPoints and then reassembling them into a pubnonce encoding. Intuitively I'd expect the `u256 -> ECPoint` conversion to be an already existing function that got reused here. It's not a huge deal since it's probs a 3 line fn and this method is still small enough to reason about.